test/fakeintake/server/parser.go

```diff
@@ -0,0 +1,83 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package server
+
+import (
+	"errors"
+	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
+	"github.com/DataDog/datadog-agent/test/fakeintake/api"
+)
+
+type PayloadParser struct {
+	payloadJsonStore map[string][]api.ParsedPayload
+	parserMap        map[string]func(api.Payload) (interface{}, error)
+}
+
+func NewPayloadParser() PayloadParser {
+	parser := PayloadParser{
+		payloadJsonStore: map[string][]api.ParsedPayload{},
+		parserMap:        map[string]func(api.Payload) (interface{}, error){},
+	}
+	parser.parserMap["/api/v2/logs"] = parser.getLogPayLoadJson
+	parser.parserMap["/api/v2/series"] = parser.getMetricPayLoadJson
+	parser.parserMap["/api/v1/check_run"] = parser.getCheckRunPayLoadJson
+	return parser
+}
+
+func (fi *PayloadParser) getJsonPayload(route string) ([]api.ParsedPayload, error) {
+	payload, ok := fi.payloadJsonStore[route]
+	if ok {
+		return payload, nil
+	}
+	return nil, errors.New("invalid route")
+}
+
+func (fi *PayloadParser) isValidRoute(route string) bool {
```

Comment:
What do you think about naming the function like `IsRouteHandled` as `valid` can mean the route is malformed?